src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -230,21 +230,28 @@ public boolean populateFare(Fare fare,
                                    List<Ride> rides,
                                    Collection<FareRuleSet> fareRules
     ) {
+        long freeTransferTripTime = 0;
         float cost = 0;
         float orcaFareDiscount = 0;
         for (Ride ride : rides) {
             RideType rideType = classify(ride.routeData);
+            freeTransferTripTime += ride.endTime - ride.startTime;
```

Comment:
Danger, danger... I think you need to add the walk times and wait times between each leg, as they are part of the two-hour window from when riders first use the card, right?

Reply:
IMO the correct way would be to compute how much time has elapsed since the first boarding that resets the two-hour window we are dealing with, i.e. `ride.startTime - firstRide.startTime`, and if that is over two hours, then reset the time window and `firstRide` and begin a new fare. That way any wait times between legs are included.

Reply:
Yes, this is incorrect. The startTime alone can be used as that is when the transaction or free transfer will occur. The logic should result in allowing a free transfer if the current start time is less than 2 hours after the last know free-transfer enabled start time (so a trip starting with WSF wouldn't result in a free transfer being applied).

Reply:
Also @br648, could you add a comment stating that the computed fare for Orca card users takes into account realtime trip updates where available, so that, for instance, when a leg on a long itinerary is delayed to begin after the initial two hour window has expired, the calculated fare for that trip will be two one-way fares instead of one.